extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/transform/JsonObjectToContractRequestTransformer.java

```diff
@@ -52,31 +49,41 @@ public JsonObjectToContractRequestTransformer(Monitor monitor) {
                 .counterPartyAddress(counterPartyAddressOrConnectorAddress(jsonObject, context))
                 .protocol(transformString(jsonObject.get(PROTOCOL), context));
 
+        contractRequestBuilder.contractOffer(contractOffer(jsonObject, context));
+
+        var callbackAddress = jsonObject.get(CALLBACK_ADDRESSES);
+        if (callbackAddress != null) {
+            var addresses = new ArrayList<CallbackAddress>();
+            transformArrayOrObject(callbackAddress, CallbackAddress.class, addresses::add, context);
+            contractRequestBuilder.callbackAddresses(addresses);
```

Comment:
this was out of my jurisdiction here, anyway I used a cleaner approach. That `transformArrayOrObject` method in my opinion does not make any sense... jsonld objects are always expanded as arrays, the decision on deserialization is to go with a single object (the first) or with the full array. This could be something do be addressed in another issue